pkg/resource/alias/hooks.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package alias
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+// customUpdateAlias calls UpdateAlias function to update a an alias
+// if it changed.
+func (rm *resourceManager) customUpdateAlias(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateAlias")
+	defer exit(err)
+
+	var updated *resource
+	updated = desired
+	if delta.DifferentAt("Spec.Description") ||
+		delta.DifferentAt("Spec.FunctionName") ||
+		delta.DifferentAt("Spec.FunctionVersion") ||
+		delta.DifferentAt("Spec.RoutingConfig") {
```

Comment:
We will never get here unless something actually changed about the Alias. The Delta compare check already handles this so you should remove it here.

Reply:
Yup nice catch